zebra-rpc/src/methods.rs

```diff
@@ -562,18 +562,21 @@ where
     // - use `height_from_signed_int()` to handle negative heights
     //   (this might be better in the state request, because it needs the state height)
     // - create a function that handles block hashes or heights, and use it in `z_get_treestate()`
-    fn get_block(&self, height: String, verbosity: u8) -> BoxFuture<Result<GetBlock>> {
+    fn get_block(&self, hash_or_height: String, verbosity: u8) -> BoxFuture<Result<GetBlock>> {
```

Comment:
Does either `lightwalletd` version send negative heights here?

If they do, we'll need to run heights through `height_from_signed_int()`. But if not, it's out of scope.

https://github.com/ZcashFoundation/zebra/blob/4fb2417adc70f57f42e9df4c8467279058b743d7/zebra-rpc/src/methods.rs#L1361-L1369

Reply:
> Does either lightwalletd version send negative heights here?

I haven't found any incidents that do, but zcashd handles -1 here so I think we should too.

This brought my attention to the `FromStr` impl for `HashOrHeight`:

https://github.com/ZcashFoundation/zebra/blob/main/zebra-chain/src/block/hash.rs#L151-L157

Now I'm wondering if there's an overlap between a valid hash vs height in string format.

Reply:
Valid heights are 10 or 18 decimal digits, valid hashes are 64 hex digits.

Reply:
> I haven't found any incidents that do, but zcashd handles -1 here so I think we should too.

I agree, I already put a TODO on those RPCs. We can do it when someone needs it, or when we're making other changes to those RPCS.